File: src/main/java/com/eager/questioncloud/mail/EmailVerification.java
package com.eager.questioncloud.mail;

import java.util.UUID;
import lombok.Builder;
import lombok.Getter;

@Getter
public class EmailVerification {
    private String token;
    private Long uid;
    private EmailVerificationType emailVerificationType;
    private Boolean isVerified;

    @Builder
    public EmailVerification(String token, Long uid, EmailVerificationType emailVerificationType, Boolean isVerified) {
        this.token = token;
        this.uid = uid;
        this.emailVerificationType = emailVerificationType;
        this.isVerified = isVerified;
    }

    public void verify() {
        this.isVerified = true;
    }

    public static EmailVerification create(Long uid, EmailVerificationType emailVerificationType) {
        return EmailVerification.builder()
            .token(UUID.randomUUID().toString())
            .uid(uid)
            .emailVerificationType(emailVerificationType)
            .isVerified(false)
            .build();
    }

    public EmailVerificationEntity toEntity() {
        return EmailVerificationEntity.builder()
            .token(token)
            .uid(uid)
            .emailVerificationType(emailVerificationType)
            .isVerified(isVerified)
            .build();
    }
}

File: qc-core/qc-user/qc-user-core/src/main/kotlin/com/eager/questioncloud/user/domain/EmailVerification.kt
package com.eager.questioncloud.user.domain

import com.eager.questioncloud.common.mail.Email
import com.eager.questioncloud.user.enums.EmailVerificationType
import com.eager.questioncloud.user.template.EmailVerificationTemplateCreator
import java.util.*

class EmailVerification(
    val token: String = UUID.randomUUID().toString(),
    val resendToken: String = UUID.randomUUID().toString(),
    val uid: Long,
    val email: String,
    val emailVerificationType: EmailVerificationType,
    var isVerified: Boolean = false,
) {
    fun verify() {
        this.isVerified = true
    }
    
    companion object {
        fun create(
            uid: Long,
            email: String,
            emailVerificationType: EmailVerificationType
        ): EmailVerification {
            return EmailVerification(
                uid = uid,
                email = email,
                emailVerificationType = emailVerificationType
            )
        }
    }
    
    fun toEmail(): Email {
        val template = EmailVerificationTemplateCreator.getTemplate(emailVerificationType, token)
        return Email(email, template.title, template.content)
    }
}
